google/cloud/storage/internal/generate_message_boundary.h

```diff
@@ -58,6 +67,24 @@ template <typename RandomStringGenerator,
 std::string GenerateMessageBoundary(
     std::string const& message, RandomStringGenerator&& random_string_generator,
     int initial_size, int growth_size) {
+  // This is not the only place we make this assumption. Duplicating the static
+  // assertion should make it easier to find the code if we ever need to fix it.
+  static_assert(std::numeric_limits<unsigned char>::max() == 255,
+                "required for efficient GenerateMessageBoundary");
```

Comment:
Do we need to make the assumption?  Couldn't we size the `seen` array with `std::numeric_limits<unsigned char>::max()`?

Reply:
Would work everywhere except on Cray computers where `unsigned char` was 64-bits 😀 

This is mostly to make it easier to find the code when/if we try to port to a machine with 16-bit chars (which unlike Cray computers still exist).

Reply:
Ack.  But platforms with 16-bit `char` (still) exist?  I'm surprised.